doc/colvars-refman-main.tex

```diff
@@ -1009,42 +1012,58 @@
-Conversely, any information that the state file has about variables or biases that are no longer defined is silently ignored.
-\emph{Because these checks are performed based solely on the names of variables and biases, it is your responsibility to ensure that these names correspond to consistent definitions between runs.}
+\cvsubsubsec{Changing configuration upon restarting.}{}
 
-When restarting using binary state files, configuration changes are not allowed.  The easiest solution would be to produce a text-format state file specifically for that purpose.
-\ifdefined\cvscriptapi{%
-Alternatively, after restarting Colvars using a state file consistent with the previous configuration, the configuration may be changed using the scripting interface (see \ref{sec:cv_scripting}).
-}\fi
+In some cases, it is useful to modify the configuration of variables or biases between consecutive runs, for example by adding or removing a restraint.
+Some special provisions will happen in that case.
+When a state file is loaded, no information is available from in regarding the any newly added variable or bias, which will remain uninitialized until the first compute step.
+Conversely, any information that the state file contains about variables or biases that are no longer defined will be silently ignored.
+Because these checks are performed based only on the \emph{names} of variables and biases, it is your responsibility to ensure that these names have \emph{consistent definitions between runs.}
+
+The flexibility just described carries some limitations: namely, it is only supported when reading \emph{text-format} state files.
+Instead, restarting from binary files\cvlammpsonly{ (such as the LAMMPS restart file)}\cvgromacsonly{ (such as the GROMACS checkpoint file)} after a configuration change will trigger an error.
+It is also important to remind that when switching to a different build of \MDENGINE, the binary format may change slightly, even if the release version is the same.
+
+To work around the potential issues just described, a text-format Colvars state file should be used.
+\ifdefined\cvvmdornamd{This is the default in \MDENGINE{} unless the ``\texttt{COLVARS\_BINARY\_RESTART}'' is set to 1, and this information is only provided for troubleshooting.}\fi\cvlammpsonly{This can be achieved by providing an explicit \texttt{input} keyword when initializing the Colvars fix (see \ref{sec:colvars_mdengine_parameters}), which will instruct Colvars to use the given filename instead of the LAMMPS restart file.
+  Furthermore, the \texttt{fix\_modify} scripting command allows to load a Colvars file after initialization (\ref{sec:cv_command_loadsave}).}\cvgromacsonly{In GROMACS, this step would normally entail generating a new TPR file: however, there is no provision to embed a text-format Colvars state file in a TPR file.
```

Comment:
I would remove the sentence "In Gromacs... TPR file", which explains the rationale of the design, but is not directly useful to users. I'd go directly to telling users what to do.